libqtile/backend/wayland/xwindow.py

```diff
@@ -168,13 +139,92 @@ def _on_set_class(self, _listener: Listener, _data: Any) -> None:
             self.ftm_handle.set_app_id(self._wm_class or "")
 
     def hide(self) -> None:
-        if self.mapped:
-            self._unmapping = True
-            self.surface.unmap_event.emit()
+        super().hide()
+
+        if self.tree:
+            self.tree.node.destroy()
+            self.tree = None
+
+            # We stop listening for commit events when unmapped, as the underlying
+            # surface can get destroyed by the client.
+            self.finalize_listener(self.surface.surface.commit_event)
 
     def unhide(self) -> None:
-        if not self.mapped:
-            self.surface.map_event.emit()
+        if self not in self.core.pending_windows:
+            if self.group and self.group.screen:
+                # Only when mapping does the xwayland_surface have a wlr_surface that we can
+                # listen for commits on and create a tree for.
+                self.add_listener(self.surface.surface.commit_event, self._on_commit)
+                if not self.tree:
+                    self.tree = SceneTree.subsurface_tree_create(
+                        self.container, self.surface.surface
+                    )
+                    self.tree.node.set_position(self.borderwidth, self.borderwidth)
+
+                self.container.node.set_enabled(enabled=True)
+                self.core.focus_window(self)
```

Comment:
I still think this needs to be removed. This gives me buggy pointer focus (the emacs + firefox issue) and hide/unhide still works after this

Reply:
I'm not sure I fully understand 1) how it works without this and 2) why anything bad should happen with it, as it's basically a no-op when focussing the currently focussed window.